### PR TITLE
Not a plugin Validator

### PR DESCRIPTION
**Pull request by ChihChengLiang**
### How was it fixed?

[//]: # (For important changes, please add a new entry to the release notes file)
[//]: # (See: https://trinity-client.readthedocs.io/en/latest/contributing.html#pull-requests)

#### Cute Animal Picture

put a cute animal picture link inside the parentheses

### CODE REVIEW

**Comment by ChihChengLiang** [comment]
```
mkdir /tmp/alice
mkdir /tmp/bob
```

Terminal A
```
export PYTHONWARNINGS="ignore::DeprecationWarning"
trinity-beacon --trinity-root-dir=/tmp/alice --beacon-nodekey=6b94ffa2d9b8ee85afb9d7153c463ea22789d3bbc5d961cc4f63a41676883c19 -l debug
```

Terminal B
```
export PYTHONWARNINGS="ignore::DeprecationWarning"
trinity-beacon --trinity-root-dir=/tmp/bob --port=556 --beacon-nodekey=f5ad1c57b5a489fc8f21ad0e5a19c1f1a60b8ab357a2100ff7e75f3fa8a4fd2e --bootstrap_nodes=enode://c289557985d885a3f13830a475d649df434099066fbdc840aafac23144f6ecb70d7cc16c186467f273ad7b29707aa15e6a50ec3fde35ae2e69b07b3ddc7a36c7@0.0.0.0:30303  -l debug
```

**Comment by mhchia** [comment]
Add a script `run_beacon_nodes.sh` to run two nodes *ALICE* and *BOB*.

### Usage
```
$ /path/to/run_beacon_nodes.sh
```
Ctrl+C to stop the nodes. (`kill -9` is used here, which is nasty.)

### Config
https://github.com/ethereum/trinity/blob/f87dfe91b398142d0892c00c55d5a4674b91f1df/eth2/beacon/scripts/run_beacon_nodes.sh#L3-L11

Note that `TIME_BOB_AFTER_ALICE` is "how long to wait to launch *BOB* after *ALICE* is launched".

### Side effects
- `DIR_ALICE` and `DIR_BOB` are removed and recreated by the script.
- Outputs are not handled, so the output of *ALICE* and *BOB* are mixed. You can check the logs manually in `{DIR_ALICE}/mainnet/logs-beacon/trinity.log` and `{DIR_BOB}/mainnet/logs-beacon/trinity.log`.

**Comment by mhchia** [comment]
Rewrite `run_beacon_nodes.sh` with Python, and name it `run_beacon_nodes.py`.

Usage is similar.
```bash
$ ./run_beacon_nodes.py
```

Logs are prefixed with the node's name and its stdout/stderr interleavingly.
```bash
Alice.stderr:     INFO  04-01 15:38:47              Endpoint  EventBus Endpoint bbeacon-node connecting to other Endpoint discovery
Alice.stderr:     INFO  04-01 15:38:47     BeaconChainSyncer  No suitable peers to sync with: Not connected to anyone
Bob.stderr:    DEBUG  04-01 15:38:48               trinity  Logging initialized: PID=80738
Bob.stderr:     INFO  04-01 15:38:48               trinity 
```

Kill the nodes with SIGINT(Ctrl+C)
```bash
^CStopping proc=<Process 80722>
Stopping proc=<Process 80738>
```

File: trinity/plugins/eth2/beacon/plugin.py [hunk 1]
```diff
@@ -73,12 +146,174 @@ def do_start(self) -> None:
...
+            state,
+            # TODO: Change back to `slot` instead of `slot + 1`.
+            # Currently `apply_state_transition_without_block` only returns the post state
+            # of `slot - 1`, so we increment it by one to get the post state of `slot`.
+            Slot(slot + 1),
+        )
+        self.logger.debug(
+            bold_green(f"Skip block, post state={post_state.root.hex()}")
+        )
+        # FIXME: We might not need to persist state for skip slots since `create_block_on_state`
```
Comment (ralexstokes) [inline trinity/plugins/eth2/beacon/plugin.py]: just to add some context, `apply_state_transition_without_block` is more of a "utility" method, and less expected to be used in application code

so i would agree w/ this `FIXME1` :)

**Comment by ChihChengLiang** [comment]
All these changes are reflected in master now.

**Pull request closed without merging**